### PR TITLE
Add route-dependant speed sections

### LINKED ISSUE #6519: Speed section by routes implement new schema
### Description and goal

- Adapt  railjson schema
- Adapt editoast
- Adapt core (only parser)
- Adapt railjson generator
- Bump railjson version
- Check website documentation is up to date

### PR DESCRIPTION
**Pull request by leovalais**
closes #6519

### CODE REVIEW

**Review by shenriotpro — APPROVED** [review]
OK for tests/

File: python/osrd_schemas/osrd_schemas/infra.py [hunk 1]
```diff
@@ -301,6 +301,11 @@ class SpeedSection(BaseObjectTrait):
     track_ranges: List[ApplicableDirectionsTrackRange] = Field(
         description="List of locations where speed section is applied"
     )
+    on_routes: Optional[List[Identifier]] = Field(
```
Comment (shenriotpro) [inline python/osrd_schemas/osrd_schemas/infra.py:304]: nit: maybe it should just be a list with an empty default?

**Review by flomonster — APPROVED** [review]
LGTM. Review and tested. The speed section edition seems to work properly.

File: python/osrd_schemas/osrd_schemas/infra.py (hunk 1 above)
Comment (leovalais) [reply python/osrd_schemas/osrd_schemas/infra.py:304]: According to the schema, there is a difference between null and [] (https://osrd.fr/en/docs/reference/design-docs/signaling/speed-limits/#data-model). If it turns out to be unnecessary, I'll update the schema accordingly later. Is it ok for you?

**Review by Erashin — APPROVED** [review]
LGTM

**Pull request merged**